tests/autoyast/installation.pm

```diff
@@ -55,32 +54,23 @@ sub save_logs_and_continue {
     select_console 'installation';
 }
 
-sub save_logs_in_linuxrc {
-    my $name = shift;
-    select_console 'install-shell2', tags => 'install-shell';
-
-    # save_y2logs is not present
-    assert_script_run "tar czf /tmp/logs-$name.tar.bz2 /var/log";
-    upload_logs "/tmp/logs-$name.tar.bz2";
-}
-
 sub handle_expected_errors {
-    my ($stage, %args) = @_;
+    my (%args) = @_;
     my $i = $args{iteration};
     record_info('Expected error', 'Iteration = ' . $i);
     send_key "alt-s";    #stop
-    save_logs_and_continue("stage1_error$i");
+    save_logs_and_continue("error$i");
     $i++;
     send_key "tab";      #continue
     send_key "ret";
     wait_idle(5);
 }
 
 sub run {
-    my @needles = ("bios-boot", "autoyast-error", "reboot-after-installation", "linuxrc-install-fail");
-    push @needles, "autoyast-confirm"        if get_var("AUTOYAST_CONFIRM");
-    push @needles, "autoyast-postpartscript" if get_var("USRSCR_DIALOG");
-    if (get_var("AUTOYAST_LICENSE")) {
+    my @needles = ('bios-boot', 'nonexisting-package', 'reboot-after-installation', 'linuxrc-install-fail');
```

Comment:
can be further simplified to `my @needles = qw(bios-boot nonexisting-package reboot-after-installation linuxrc-install-fail)`